litmus-portal/backend/subscriber/subscriber.go

```diff
@@ -0,0 +1,99 @@
+/*
+	----------------------Subscriber Logic---------------------------------
+	Step1: Check if cluster is already registered or not from the configmap.
+	Step2: Register the subscriber with the admin cluster.
+	Step3: Waiting for request from the graphql server using subscription.
+	NOTE: format:
+	{
+		"requestid": Int!, // Randomly Generated id from the server.
+		"requesttype": String!, // request type can be GET, UPDATE, CREATE and DELETE.
+		"manifest": Object, // K8s manifest, Not needed when it is a GET or DELETE request.
+		"resourcetype": String!, // It can be pre defined or custom k8s resource.
+		"namespace": "default" // namespace of the k8s resource.
+	}
+	! stands for required fields.
+	Step4: Call the the clusterOperation function using the above format.
+	Step5: Response back to the graphql Server.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"github.com/litmuschaos/litmus/litmus-portal/backend/subscriber/pkg/cluster"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+)
+
+var GRAPHQL_SERVER_ADDRESS = os.Getenv("SERVER_ADDRESS") // Format: http://IP:PORT/query
+
+var kubeconfig *string
+
+func init() {
+	if GRAPHQL_SERVER_ADDRESS == "" {
+		panic("GRAPHQL SERVER ADDRESS NOT FOUND")
+	}
+}
+
+func main() {
+
+	kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+
+	client := &http.Client{}
+	query := `{
+			"query": "subscription { clusterSubscription { id data }  }"
+		}`
+
+	req, err := http.NewRequest("POST", GRAPHQL_SERVER_ADDRESS, strings.NewReader(query))
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Headers for the calling the server endpoint
+	req.Header.Set("Accept-Encoding", "gzip, deflate, br")
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Accept", "application/json")
+	req.Header.Set("Connection", "keep-alive")
+	req.Header.Set("DNT", "1")
+	req.Header.Set("Origin", "http://localhost:8080")
+
+	for {
+		resp, err := client.Do(req)
+		if err != nil {
+			log.Fatal(err)
```

Comment:
This has to be a HTTP call not a websocket, because URL package is not contain any query parameter. https://godoc.org/net/url#URL

And we don't have to worry about the close connection because its a API call not a database connection.

Reply:
cc @gdsoumya

Reply:
And this is not an infinite loop, as it waits for the response. After it gets a response from the graphql server then it goes into a second loop

Reply:
But it will get some error or timeout as the server won't respond every time?

Reply:
Does the request not timeout on the client side? The server will have an open connection but go-client should time out after some time right?

Reply:
If the server goes down, it will automatically timeout